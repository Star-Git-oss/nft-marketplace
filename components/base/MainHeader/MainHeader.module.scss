@import "style/colors.scss";


.Logo {
  width: 140px;
  cursor: pointer;
}

@media (max-width: 1300px) {
  .Logo {
    width: 120px;
    cursor: pointer;
  }
}

@media (max-width: 1000px) {
  .Logo {
    width: 200px;
    cursor: pointer;
  }
}

@media (max-width: 730px) {
  .Logo {
    width: 150px;
    cursor: pointer;
  }
}
