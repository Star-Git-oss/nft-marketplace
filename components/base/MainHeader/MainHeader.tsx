import React, { useState } from 'react'
import Link from 'next/link'
import styled from 'styled-components'

import { ModalWallet } from 'components/base/Modal'
import { ProfileMenuBadge, ProfileMenuDropdown } from 'components/base/ProfileMenu'
import Button, { AnchorButton } from 'components/ui/Button'
import Icon from 'components/ui/Icon'
import { Container, Wrapper } from 'components/layout'

import { computeCaps } from 'utils/strings'
import { useApp } from 'redux/hooks'

const MainHeader: React.FC = () => {
  const [isModalWalletExpanded, setIsModalWalletExpanded] = useState(false)
  const [isProfileMenuExpanded, setIsProfileMenuExpanded] = useState(false)

  const { user } = useApp()

  const isNftCreationEnabled =
    process.env.NEXT_PUBLIC_IS_NFT_CREATION_ENABLED === undefined
      ? true
      : process.env.NEXT_PUBLIC_IS_NFT_CREATION_ENABLED === 'true'

  return (
    <>
      <Container>
        <SWrapper>
          <Link href="/">
            <a title="Marketplace homepage">
              <SLogo name="logoTernoaBlack" />
            </a>
          </Link>
          <SNavContainer>
            <SNavLinksContainer>
              <Link href="/explore" passHref>
                <SLinkItem>Explore</SLinkItem>
              </Link>
              <Link href="/faq" passHref>
                <SLinkItem>How it works</SLinkItem>
              </Link>
            </SNavLinksContainer>
            <SNavButtonsCointainer>
              {isNftCreationEnabled && (
                <Link href="/create" passHref>
                  <>
                    <AnchorButton color="contrast" href="/create" size="medium" text="Create NFT" variant="outlined" />
                  </>
                </Link>
              )}
              {user === undefined || user === null || user._id === '' ? (
                <Button
                  color="contrast"
                  onClick={() => setIsModalWalletExpanded(true)}
                  size="medium"
                  text="Connect"
                  variant="outlined"
                />
              ) : (
                <ProfileMenuBadge
                  onClick={() => setIsProfileMenuExpanded((prevState) => !prevState)}
                  tokenAmount={user.capsAmount ? computeCaps(Number(user.capsAmount)) : 0}
                  tokenSymbol="CAPS"
                  user={user}
                />
              )}
            </SNavButtonsCointainer>
          </SNavContainer>
          {user && isProfileMenuExpanded && (
            <SProfileMenuDropdown onClose={() => setIsProfileMenuExpanded(false)} user={user} />
          )}
        </SWrapper>
      </Container>
      {isModalWalletExpanded && <ModalWallet setExpanded={setIsModalWalletExpanded} />}
    </>
  )
}

const SWrapper = styled(Wrapper)`
  flex-direction: row;
  justify-content: center;
  position: relative;
  padding-top: 3.2rem !important;
  padding-bottom: 3.2rem !important;

  ${({ theme }) => theme.mediaQueries.lg} {
    justify-content: space-between;
    padding-top: 4rem !important;
    padding-bottom: 4rem !important;
  }
`

const SLogo = styled(Icon)`
  width: 16rem;
  cursor: pointer;
`

const SNavContainer = styled.div`
  display: none;

  ${({ theme }) => theme.mediaQueries.lg} {
    display: flex;
    align-items: center;
  }
`

const SNavLinksContainer = styled.div`
  display: flex;
  align-items: center;

  > * {
    &:not(:first-child) {
      margin-left: 2.4rem;
    }
  }
`

const SLinkItem = styled.a`
  color: ${({ theme }) => theme.colors.contrast};
  cursor: pointer;
  font-family: ${({ theme }) => theme.fonts.bold};
  font-size: 1.6rem;
`

const SNavButtonsCointainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 2.4rem;

  ${({ theme }) => theme.mediaQueries.lg} {
    margin-left: 5.6rem;
  }
`

const SProfileMenuDropdown = styled(ProfileMenuDropdown)`
  top: 10.4rem;
  right: 3.2rem;

  &::before {
    width: 0;
    height: 0;
    border-left: 2.4rem solid transparent;
    border-right: 2.4rem solid transparent;
    z-index: 101;
    border-top: ${({ theme }) => `1.2rem solid ${theme.colors.invertedContrast}`};
    content: '';
    position: absolute;
    top: -1.6rem;
    right: 3.2rem;
    transform: translateY(0.8rem) rotate(180deg);
  }
`

export default MainHeader
