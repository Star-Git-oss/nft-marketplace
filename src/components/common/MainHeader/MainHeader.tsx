import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import styled from 'styled-components';
import { useTranslation } from 'react-i18next';

import { FaSearch } from 'react-icons/fa';
import Col from '../ui-library/Col/Col';
import Row from '../ui-library/Row/Row';
import Button from '../ui-library/Button/Button';
import Input from '../ui-library/Input/Input';

import { H3 } from '../Title/Title';
import { ReactComponent as TernoaLogo } from '../assets/logo-ternoa.svg';
import { ReactComponent as Caps } from '../assets/caps.svg';

const ContainerHeader = styled.div`
  padding:21px 0;
  width:100%;
`;

const TitleHeader = styled(H3)`
  text-align:right;
  margin:0;
  padding:0;
  cursor: pointer;
  outline: none;
`;

const SearchStyled = styled.div`
  position: relative;
  display: block;
  margin: 0px auto;
  text-align: center;
  z-index: 2;
`;

const FaSearchStyled = styled(FaSearch)`
  position: absolute;
  top: 12px;
  left: 15px;
  font-weight: bold;
  font-style: normal;
`;

const InputStyled = styled(Input)`
  line-height: 30px;
  padding-left: 30px;
  z-index: 1;
`;

type MainHeaderType = {
  walletId: string;
};

const MainHeader: React.FC<MainHeaderType> = ({ walletId }) => {
  
  const [searchValue, setSearchValue] = useState('' as string);
  
  const history = useHistory();
  const { t } = useTranslation();

  const updateKeywordSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchValue(event.currentTarget.value);
  };
  
  return (
    <ContainerHeader>
      
      <Row>
        {/* Application name */}
        <Col small="50" medium="20" large="20">
          <TitleHeader role="button" tabIndex={0} onClick={() => { history.push('/');}}>
            <TernoaLogo />
            <span>&nbsp; Ternoa Stamp</span>
          </TitleHeader>
          
        </Col>
        {/* Search Button */}
        <Col small="50" medium="40" large="40">
          <SearchStyled>
            <FaSearchStyled />
            <InputStyled onChange={updateKeywordSearch} full placeholder={t('header.find')} type="search" />
          </SearchStyled>
        </Col>
        
        {/* Buttons menu */}
        <Col small="100" medium="40" large="40">
          <div>

            <Button 
              onClick={() => {history.push(`/search?q=${searchValue}`);}}
            >
              <FaSearch />
            </Button>

            <Button 
              primary 
              onClick={() => {history.push('/create');}}
            >
              {t('header.createButton')}
            </Button>
        
            { walletId ? 

              (
                <Button 
                  onClick={() => {history.push('/profile-top');}}
                >
                  <span>{walletId} &nbsp; <Caps /></span>
                </Button>
              )
              :
              (
                <Button 
                  onClick={() => {history.push('/connect-wallet');}}
                >
                  <span>{t('header.connectWallet')} </span>
                </Button>
              )}
              
            
          </div>
        
        </Col>
      </Row>
    </ContainerHeader>
  );
};

export default MainHeader;
