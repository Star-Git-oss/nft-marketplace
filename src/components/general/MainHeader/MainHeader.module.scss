.Header {
  width: 100%;
  margin: 0px;
  padding: 0px;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
}

.HeaderContainer {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 1400px;
  box-sizing: border-box;
  padding: 16px 40px;
  padding-top: 24px;
}

.Logo {
  width: 150px;
  cursor: pointer;
}

.Input {
  background: rgba(0, 0, 0, 0.05);
  border-radius: 30px;
  font-size: 14px;
  border: none;
  width: 430px;
  padding: 12px 24px;
  outline: none;
  font-family: 'Airbnb Cereal App Book';
}

.Infos {
  display: flex;
  align-items: center;
}

.Links {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 270px;
}

.LinkItem {
  //font-family: "Airbnb Cereal App Bold";
  font-family: 'Airbnb Cereal App Medium';
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  line-height: 17px;
  color: black;
  text-decoration: none;
}

// wallets

.Wallet {
  display: flex;
  margin-left: 80px;
}

.Regular {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Create {
  margin-right: 24px;
  font-family: 'Airbnb Cereal App Medium';
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  text-align: center;
  border-radius: 28px;
  padding: 10px 28px;
  border: 2px solid black;
  cursor: pointer;
  transition: all 0.4s ease-out;

  &:hover {
    color: white;
    background-color: black;
  }
}

.Connect {
  margin-right: 24px;
  font-family: 'Airbnb Cereal App Medium';
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  display: flex;
  align-items: center;
  text-align: center;
  border-radius: 28px;
  padding: 10px 28px;
  border: 2px solid black;
  color: white;
  background-color: black;
  cursor: pointer;
  transition: all 0.4s ease-out;

  &:hover {
    color: black;
    background-color: white;
  }
}

@media (max-width: 1300px) {
  .Links {
    display: none;
  }

  .HeaderContainer {
    justify-content: space-around;
    width: 100%;
  }

  .Wallet {
    margin-left: 0px;
  }

  .Logo {
    width: 120px;
    cursor: pointer;
  }
}
