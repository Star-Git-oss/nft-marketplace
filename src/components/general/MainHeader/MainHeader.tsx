import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

import Logo from '../assets/LogoTernoa';

import style from './MainHeader.module.scss';

const MainHeader: React.FC<any> = () => {
  const [searchValue, setSearchValue] = useState('' as string);

  const history = useHistory();
  const { t } = useTranslation();

  const walletId = null;

  const updateKeywordSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
    setSearchValue(event.currentTarget.value);
  };

  return (
    <div className={style.Header}>
      <div className={style.HeaderContainer}>
        <Logo className={style.Logo} onClick={(e) => true} />
        <div className={style.SearchBar}>
          <input
            type="search"
            onChange={updateKeywordSearch}
            className={style.Input}
            placeholder="Search"
          />
        </div>
        <div className={style.Infos}>
          <div className={style.Links}>
            <a className={style.LinkItem} href="#">
              Explore
            </a>
            <a className={style.LinkItem} href="#">
              How it works ?
            </a>
            <a className={style.LinkItem} href="#">
              Support
            </a>
          </div>
          <div className={style.Wallet}>
            {walletId ? (
              <div>connected</div>
            ) : (
              <div className={style.Regular}>
                <div className={style.Create}>Create NFT</div>
                <div className={style.Connect}>Connect</div>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default MainHeader;

/*

<ContainerHeader>

            <Row>

                <Col small="50" medium="20" large="20">
                    <TitleHeader role="button" tabIndex={0} onClick={() => { history.push('/'); }}>
                        <TernoaLogo />
                        <span>&nbsp; Ternoa Stamp</span>
                    </TitleHeader>

                </Col>

                <Col small="50" medium="40" large="40">
                    <SearchStyled>
                        <FaSearchStyled />
                        <InputStyled onChange={updateKeywordSearch} full placeholder={t('header.find')} type="search" />
                    </SearchStyled>
                </Col>


                <Col small="100" medium="40" large="40">
                    <div>

                        <Button
                            onClick={() => { history.push(`/search?q=${searchValue}`); }}
                        >
                            <FaSearch />
                        </Button>

                        <Button
                            primary
                            onClick={() => { history.push('/create'); }}
                        >
                            {t('header.createButton')}
                        </Button>

                        {walletId ?

                            (
                                <Button
                                    onClick={() => { history.push('/profile-top'); }}
                                >
                                    <span>{walletId} &nbsp; <Caps /></span>
                                </Button>
                            )
                            :
                            (
                                <Button
                                    onClick={() => { history.push('/connect-wallet'); }}
                                >
                                    <span>{t('header.connectWallet')} </span>
                                </Button>
                            )}


                    </div>

                </Col>
            </Row>
        </ContainerHeader>



*/
