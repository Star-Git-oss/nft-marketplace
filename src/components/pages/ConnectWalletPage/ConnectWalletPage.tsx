/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import styled from 'styled-components';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import { H1, P } from '../../common/Title/Title';
import TinyContainer from '../../common/ui-library/TinyContainer/TinyContainer';
import Button from '../../common/ui-library/Button/Button';

const WalletButton = styled(Button)`
  width: 200px;
  margin:10px;
  & i img {
    width: 20px;
  }
`;

const walletsList = [
  { id: 1, label: 'Metamask', img: '/wallets/Metamask.png' },
  { id: 2, label: 'Wallet Connect', img: '/wallets/WalletConnect.png' },
  { id: 3, label: 'Fortmatic', img: '/wallets/Fortmatic.png' },
  { id: 4, label: 'WalletLink', img: '/wallets/WalletLink.png' },
];

type ConnectWalletPageType = {
  setWalletId: (walletId: string) => void;
};

const ConnectWalletPage: React.FC<ConnectWalletPageType> = (props) => {

  const { t } = useTranslation();

  const connectWallet = () => {
    props.setWalletId('12,450');
  };

  return (
    <TinyContainer style={{ textAlign: 'center' }}>

      <Helmet>
        <title>{t('walletConnection.seo.title')}</title>
        <meta name="description" content={t('walletConnection.seo.description')} />
        <meta name="keywords" content={t('walletConnection.seo.keywords')} />
      </Helmet>

      <H1>{t('walletConnection.title')}</H1>

      <P>
        {t('walletConnection.introduction1')}
        <br />
        {t('walletConnection.introduction2')}
      </P>

      <div>
        {walletsList.map( (wallet) => (
          <>
            <WalletButton onClick={connectWallet} key={wallet.id}>
              <i>
                <img alt="wallet" src={wallet.img} />
              </i>
              {wallet.label}
            </WalletButton>
            <br />
          </>
        ))}
      </div>

    </TinyContainer>
  );
};

export default ConnectWalletPage;