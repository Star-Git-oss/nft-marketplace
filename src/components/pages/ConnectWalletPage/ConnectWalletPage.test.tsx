/* eslint-disable no-console */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ConnectWalletPage from './ConnectWalletPage';
import MainHeader from '../../common/MainHeader/MainHeader';
import ContextProvider from '../../../utils/store/store';

// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

jest.mock('../../../utils/store/actions', () => ({
  updateStoreElement: jest.fn(),
}));

describe('ConnectWalletPage snapshots', () => {

  test('ConnectWalletPage renders correctly', () => {
    const tree = renderer
      .create(<ConnectWalletPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('ConnectWalletPage', () => {

  it('renders component properly', () => {
    render(<ConnectWalletPage />);
    expect(screen.getByText('Metamask')).toBeInTheDocument();
    expect(screen.getByText('Wallet Connect')).toBeInTheDocument();
    expect(screen.getByText('Fortmatic')).toBeInTheDocument();
    expect(screen.getByText('WalletLink')).toBeInTheDocument();
  });

  it('Show wallet amount after clicking', () => {
    render(
      <ContextProvider>
        <MainHeader />
        <ConnectWalletPage />
      </ContextProvider>
    );

    const metaMask = screen.getByText('Metamask');
    expect(metaMask).toBeInTheDocument();
    fireEvent.click(metaMask);
    screen.findByText('12,450');

  });

  
});