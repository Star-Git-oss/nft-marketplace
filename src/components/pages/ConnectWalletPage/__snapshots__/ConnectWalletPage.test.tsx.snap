// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWalletPage snapshots ConnectWalletPage renders correctly 1`] = `
<div
  className="sc-jrAGrp fBvUYs"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1
    className="sc-dlfnbm jnRgnW"
  >
    walletConnection.title
  </h1>
  <p
    className="sc-iBPRYJ kEpCMi"
  >
    walletConnection.introduction1
    <br />
    walletConnection.introduction2
  </p>
  <div>
    <button
      className="sc-kEjbxe jGFMIo sc-iqHYGH cxgTKu"
      onClick={[Function]}
    >
      <i>
        <img
          alt="wallet"
          src="/wallets/Metamask.png"
        />
      </i>
      Metamask
    </button>
    <br />
    <button
      className="sc-kEjbxe jGFMIo sc-iqHYGH cxgTKu"
      onClick={[Function]}
    >
      <i>
        <img
          alt="wallet"
          src="/wallets/WalletConnect.png"
        />
      </i>
      Wallet Connect
    </button>
    <br />
    <button
      className="sc-kEjbxe jGFMIo sc-iqHYGH cxgTKu"
      onClick={[Function]}
    >
      <i>
        <img
          alt="wallet"
          src="/wallets/Fortmatic.png"
        />
      </i>
      Fortmatic
    </button>
    <br />
    <button
      className="sc-kEjbxe jGFMIo sc-iqHYGH cxgTKu"
      onClick={[Function]}
    >
      <i>
        <img
          alt="wallet"
          src="/wallets/WalletLink.png"
        />
      </i>
      WalletLink
    </button>
    <br />
  </div>
</div>
`;
